### PR TITLE
Add SIG CLI Annual Report 2023

### LINKED ISSUE #7767: 2023 Annual Report: SIG CLI
2023 Annual Report: SIG CLI

Chairs: @KnVerey @eddiezane @mpuckett159 
Liaison: @pacoxu

Actions for the chair/organizer of the community group:
- [ ] Consult your community group to complete draft of report
- [ ] If needed, consult with Steering Committee liaison on any private issues or concerns that arrise while completing report
- [ ] Submit PR to git.k8s.io/community repo with copy of report, assigning your Steering Committee liaison for review
- [ ] Steering Committee reviews report, and may make comments/ask questions
- [ ] Once all comments are addressed, Steering Committee will approve report to merge
- [ ] If needed, any follow up items may be brought to the "Chairs, Tech Leads, and Organizers" meeting in April

Once all the above items are complete, this issue may be `/close`'d

Key dates:
- Initial PR to communtiy repo should be opened by May 1, 2024
- PR should be reviewed and merged by May 22, 2024

More detailed information on the annual reports process is available [here](https://git.k8s.io/community/committee-steering/governance/annual-reports.md).

Your group's report template is available here: https://git.k8s.io/community/sig-cli/annual-report-2023.md

If you have any questions or concerns about this process, you may reach the Steering Committee via the following methods:
- Slack/E-mail your liaison directly
- The public #steering-committee channel in Slack
- The public mailing list steering@kubernetes.io
- The private mailing list steering-private@kubernetes.io for any private or sensitive issues.

/assign @KnVerey @eddiezane @mpuckett159 
/sig cli
/committee steering
/area annual-reports

### PR DESCRIPTION
**Pull request by ardaguclu**
**Which issue(s) this PR fixes**:
Fixes #7767

### CODE REVIEW

**Review by mpuckett159 — APPROVED** [review]
/lgtm thanks Arda

**Comment by MadhavJivrajani** [comment]
/hold
just to prevent accidental merge

File: sig-cli/annual-report-2023.md [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
+    - Custom profiling support in kubectl debug
+    - Transition from SPDY to Websockets
+    - Plugin resolution for non-shadowing subcommands in kubectl promoted to beta
+    - New pruning design for kubectl apply --prune, aka ApplySet
+    - Aggregated Discovery promoted to GA
+    - kubectl debug promoted to GA
+    - New minor release for kui
+    - Krew has 242 plugins
 
 2. Are there any areas and/or subprojects that your group needs help with (e.g. fewer than 2 active OWNERS)?
```
Comment (soltysh) [inline sig-cli/annual-report-2023.md:18]: I'd call out here that we've lost 2 kustomize primary maintainers in 2023, we'll be looking for more help in 2024.

File: sig-cli/annual-report-2023.md [hunk 2]
```diff
@@ -70,8 +72,8 @@ Operational tasks in [sig-governance.md]:
 - [ ] [CONTRIBUTING.md] reviewed for accuracy and updated if needed
 - [ ] Other contributing docs (e.g. in devel dir or contributor guide) reviewed for accuracy and updated if needed
 - [ ] Subprojects list and linked OWNERS files in [sigs.yaml] reviewed for accuracy and updated if needed
-- [ ] SIG leaders (chairs, tech leads, and subproject leads) in [sigs.yaml] are accurate and active, and updated if needed
-- [ ] Meeting notes and recordings for 2023 are linked from [README.md] and updated/uploaded if needed
+- [x] SIG leaders (chairs, tech leads, and subproject leads) in [sigs.yaml] are accurate and active, and updated if needed
+- [x] Meeting notes and recordings for 2023 are linked from [README.md] and updated/uploaded if needed
```
Comment (soltysh) [inline sig-cli/annual-report-2023.md:76]: Check all of the above.

File: sig-cli/annual-report-2023.md (hunk 1 above)
Comment (ardaguclu) [reply sig-cli/annual-report-2023.md:18]: Updated with this. Thanks for review.

**Review by soltysh — APPROVED** [review]
/lgtm

**Review by pacoxu — APPROVED** [review]
/lgtm

**Review by eddiezane — APPROVED** [review]
/lgtm

Ty!

**Comment by pacoxu** [comment]
Feel free to unhold as all sig TL and chairs LGTMed.

**Pull request merged**